core/trino-main/src/main/java/io/trino/execution/buffer/SpoolingExchangeOutputBuffer.java

```diff
@@ -272,6 +272,14 @@ private void updatePeakMemoryUsage(long bytes)
         }
     }
 
+    private void forceFreeMemory()
+    {
+        LocalMemoryContext context = getSystemMemoryContextOrNull();
+        if (context != null) {
+            context.close();
```

Comment:
not every memory context is closeable, see https://github.com/trinodb/trino/pull/13213/files#diff-1774b86bf4c8b2248f02693fe0b0c6acc38305a6773ca813f5957601207a8168R338.
I think we can assume that only memory context created from `io.trino.memory.context.AggregatedMemoryContext#newAggregatedMemoryContext` are closeable

Reply:
The other output buffers close the memory context through the `OutputBufferMemoryManager`: https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/execution/buffer/OutputBufferMemoryManager.java#L199

The memory context is closeable. It is created in https://github.com/trinodb/trino/blob/master/core/trino-main/src/main/java/io/trino/operator/TaskContext.java#L157 through https://github.com/trinodb/trino/blob/master/lib/trino-memory-context/src/main/java/io/trino/memory/context/MemoryTrackingContext.java#L136